src/EventListener/DataContainer/NewsCategoryListener.php

```diff
@@ -225,7 +233,26 @@ public function onGenerateAlias($value, DataContainer $dc)
         // Generate alias if there is none
         if (!$value) {
             $autoAlias = true;
-            $value = StringUtil::generateAlias($dc->activeRecord->frontendTitle ?: $dc->activeRecord->title);
+            $title = $dc->activeRecord->frontendTitle ?: $dc->activeRecord->title;
+
+            if (null !== $this->slug) {
+                /** @var Config $configAdapter */
+                $configAdapter = $this->framework->getAdapter(Config::class);
```

Comment:
I think you can safely use the `Config` class directly. `$this->framework` legacy stuff will be dropped in the future anyway as extension has no unit tests and it makes no sense to use it.

Reply:
Agreed :)